lib/workers/global/limits.ts

```diff
@@ -37,3 +38,163 @@ export function isLimitReached(key: Limit): boolean {
   const { max, current } = limit;
   return max - current <= 0;
 }
+
+export type CountName = 'ConcurrentPRs' | 'HourlyPRs' | 'Branches';
+
+type BranchLimitName =
+  | 'branchConcurrentLimit'
+  | 'prConcurrentLimit'
+  | 'prHourlyLimit';
+
+export const counts = new Map<CountName, number>();
+
+export function getCount(key: CountName): number {
+  const count = counts.get(key);
+  // istanbul ignore if: should not happen
+  if (!count) {
+    logger.warn(`Could not compute the count of ${key}, returning zero.`);
+    return 0;
+  }
+  return count;
+}
+
+export function setCount(key: CountName, val: number): void {
+  const count = val;
+  counts.set(key, count);
+  logger.debug(`${key} count = ${count}`);
+}
+
+export function incCountValue(key: CountName, incBy = 1): void {
+  const count = getCount(key);
+  counts.set(key, count + incBy);
+}
+
+function handleConcurrentLimits(
+  key: Exclude<CountName, 'HourlyPRs'>,
+  config: BranchConfig,
+): boolean {
+  const limitKey =
+    key === 'Branches' ? 'branchConcurrentLimit' : 'prConcurrentLimit';
+
+  // calculate the limits for this branch
+  const hourlyLimit = calcLimit(config.upgrades, 'prHourlyLimit');
+  const limitValue = calcLimit(config.upgrades, limitKey);
+
+  const hourlyPrCount = getCount('HourlyPRs');
+  const currentCount = getCount(key);
+
+  // if a limit is defined ( >0 ) and limit reached return true ie. limit has been reached
+  if (hourlyLimit && hourlyPrCount >= hourlyLimit) {
+    return true;
+  }
+
+  if (limitValue && currentCount >= limitValue) {
+    return true;
+  }
```

Comment:
```suggestion
  const hourlyLimit = calcLimit(config.upgrades, 'prHourlyLimit');
  const hourlyPrCount = getCount('HourlyPRs');

  // if a limit is defined ( >0 ) and limit reached return true ie. limit has been reached
  if (hourlyLimit && hourlyPrCount >= hourlyLimit) {
    return true;
  }

  const limitValue = calcLimit(config.upgrades, limitKey);
  const currentCount = getCount(key);

  if (limitValue && currentCount >= limitValue) {
    return true;
  }
```

perf only compute second limit if first isn't reached